Source/WebCore/loader/cache/CachedResourceLoader.cpp

```diff
@@ -747,6 +757,7 @@ bool CachedResourceLoader::canRequestInContentDispositionAttachmentSandbox(Cache
         return true;
     }
 
+    ASSERT(document);
     if (!document->shouldEnforceContentDispositionAttachmentSandbox() || document->securityOrigin().canRequest(url, OriginAccessPatternsForWebProcess::singleton()))
         return true;
```

Comment:
Should also do a `document` `nullptr` check here, not just assert.

```suggestion
    ASSERT(document);
    if (document && (!document->shouldEnforceContentDispositionAttachmentSandbox() || document->securityOrigin().canRequest(url, OriginAccessPatternsForWebProcess::singleton())))
        return true;
```

Reply:
Also need a `document` check in the code below (not part of the original patch):

```
    if (document) {
        String message = "Unsafe attempt to load URL " + url.stringCenterEllipsizedToLength() + " from document with Content-Disposition: attachment at URL " + document->url().stringCenterEllipsizedToLength() + ".";
        document->addConsoleMessage(MessageSource::Security, MessageLevel::Error, message);
    }
```

Reply:
Will do. I think I'll check and early return, just like the early part of this method.